barcodes_generator_abstract/models/barcode_generate_mixin.py

```diff
@@ -36,13 +36,16 @@ class BarcodeGenerateMixin(models.AbstractModel):
     @api.model
     def create(self, vals):
         """It creates a new barcode if automation is active."""
-        barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
-        if barcode_rule.exists():
-            vals.update({
-                'barcode_rule_id': barcode_rule.id,
-            })
+        automatic = False
+        if not vals.get("barcode") and not vals.get("barcode_rule_id"):
+            barcode_rule = self.env['barcode.rule'].get_automatic_rule(self._name)
+            if barcode_rule.exists():
+                automatic = True
+                vals.update({
+                    'barcode_rule_id': barcode_rule.id,
+                })
         record = super(BarcodeGenerateMixin, self).create(vals)
-        if barcode_rule:
+        if automatic:
```

Comment:
```suggestion
        if barcode_rule.exists():
```

Reply:
`False.exists()` does not exist. Is `exists()` pertinent here?